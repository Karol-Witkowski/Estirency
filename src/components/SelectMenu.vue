<template>
  <section>
    <header>
      <h3>menu</h3>
    </header>
    <h4>Select currencies and fill input to show data</h4>
    <h4>select base currency</h4>
    <v-select
      @input="setBaseCurrency"
      label="name"
      :options="this.$store.state.currency
        .filter((currencies) => currencies !== this.$store.state.targetCurrency)"
      :value="$store.state.baseCurrency"
    >
      <template #search="{attributes, events}">
        <input
          class="vs__search"
          @keypress="currencyRestrictions"
          v-bind="attributes"
          v-on="events"
        />
      </template>
    </v-select>
    <figure>
      <img
        alt="arrays icon"
        data-cy='swapIcon'
        src="@/assets/swap-vertical.png"
        v-on:click="swapValues"
      >
    </figure>
    <h4>select target currency</h4>
    <v-select
      @input="setTargetCurrency"
      label="name"
      :options="this.$store.state.currency
        .filter((currencies) => currencies !== this.$store.state.baseCurrency)"
      :value="$store.state.targetCurrency"
    >
      <template #search="{attributes, events}">
        <input
          class="vs__search"
          @keypress="currencyRestrictions"
          v-bind="attributes"
          v-on="events"
        />
      </template>
    </v-select>
    <h4>type amount</h4>
    <input
      class="amountInput"
      @input="setAmountValue"
      @keypress="amountRestrictions"
      placeholder="amount of currency"
      :value="$store.state.amountValue"
    />
  </section>
</template>

<script>
import Vue from 'vue';
import vSelect from 'vue-select';

Vue.component('v-select', vSelect);

export default {
  name: 'SelectMenu',
  components: {
    vSelect,
  },

  data() {
    return {
      baseCurrency: String,
      targetCurrency: String,
      cur1: 'PLN',
      cur2: 'EUR',
    };
  },

  methods: {
    setBaseCurrency(val) {
      if (val !== null) {
        this.$store.commit('setBaseCurrency', val);
        this.$store.state.loaded = false;
      }
    },

    setTargetCurrency(val) {
      if (val !== null) {
        this.$store.commit('setTargetCurrency', val);
        this.$store.state.loaded = false;
      }
    },

    setAmountValue(val) {
      this.$store.commit('setAmountValue', val.target.value);
    },

    currencyRestrictions(e) {
      const keyCode = e.key.charCodeAt(0);
      if (keyCode > 47 && keyCode < 58) {
        e.preventDefault();
      }
    },

    swapValues() {
      [this.$store.state.baseCurrency, this.$store.state.targetCurrency] = [
        this.$store.state.targetCurrency, this.$store.state.baseCurrency];
      this.$store.state.loaded = false;
    },

    amountRestrictions(e) {
      const keyCode = e.key.charCodeAt(0);
      if ((keyCode < 48 || keyCode > 57) && keyCode !== 46) {
        e.preventDefault();
      }

      if (keyCode === 46 && (this.$store.state.amountValue.toString().split('.').length === 2)) {
        e.preventDefault();
      }

      if (this.$store.state.amountValue != null && this.$store.state.amountValue.toString().indexOf('.')
       > -1 && (this.$store.state.amountValue.split('.')[1].length > 1)) {
        e.preventDefault();
      }
    },
  },
};
</script>

<style lang="scss">
@import '@/stylesheets/SelectMenu.scss';
@import '@/stylesheets/vSelect.scss';
</style>
