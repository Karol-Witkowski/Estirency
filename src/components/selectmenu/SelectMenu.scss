.select {
  @include card;
  font-size: 1.1em;

  h3 {
    margin-bottom: 9%;
    margin-top: 11%;
  }

  h4 {
    margin-bottom: 4.5%;
  }

  h4:first-of-type {
    margin-bottom: 14%;
    padding: 0% 3%;
    text-transform: none;
  }

  h4:last-of-type {
    margin-top: 15%;
  }

  img {
    cursor: pointer;
    display: block;
    margin: 8% auto;
    width: 50px;
  }

  .amountInput {
    background-color: transparent;
    border: $panelStyle;
    box-shadow: $primaryShadow;
    color: $secondaryColor;
    height: 5.4%;
    margin: -0.9% 9.5% auto;
    font-size: 0.88em;
    font-weight: $bold;
    font-variant: small-caps;
    text-align: center;
    width: 79%;
  }

  .amountInput:focus {
    font-size: 0.8em;
    transition: cubic-bezier(0.19, 1, 0.22, 1) 200ms;
  }
}
