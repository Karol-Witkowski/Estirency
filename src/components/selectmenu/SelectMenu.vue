<template>
    <div class="select">
      <h3>menu</h3>
      <h4>Select currencies and fill input to show data</h4>
      <h4>select base currency</h4>
      <v-select :options="$store.state.currency" label="name" :value="$store.state.baseCurrency"
      @input="setBaseCurrency" placeholder="base currency">
        <template #search="{attributes, events}">
          <input class="vs__search" v-bind="attributes" v-on="events"
          @keypress="currencyRestrictions"/>
        </template>
      </v-select>
      <figure>
        <img src="@/assets/swap-vertical.png" alt="swap currency icon" v-on:click="this.swapValues">
      </figure>
      <h4>select target currency</h4>
      <v-select :options="$store.state.currency" label="name" :value="$store.state.targetCurrency"
      @input="setTargetCurrency" placeholder="target currency">
        <template #search="{attributes, events}">
          <input class="vs__search" v-bind="attributes" v-on="events"
          @keypress="currencyRestrictions"/>
        </template>
      </v-select>
      <h4>type amount</h4>
      <input class="amountInput" @keypress="amountRestrictions" :value="$store.state.amountValue"
      @input="setAmountValue" placeholder="amount of currency"/>
    </div>
</template>

<script>
import Vue from 'vue';
import vSelect from 'vue-select';

Vue.component('v-select', vSelect);

export default {
  name: 'SelectMenu',
  components: {
    vSelect,
  },

  data() {
    return {
      cur1: 'PLN',
      cur2: 'EUR',
    };
  },

  methods: {
    amountRestrictions($event) {
      const keyCode = ($event.keyCode ? $event.keyCode : $event.which);
      if ((keyCode < 48 || keyCode > 57) && keyCode !== 46) {
        $event.preventDefault();
      }

      if ((keyCode === 46) && (this.$store.state.amountValue.toString().split('.').length === 2)) {
        $event.preventDefault();
      }

      if (this.$store.state.amountValue != null && this.$store.state.amountValue.toString().indexOf('.')
       > -1 && (this.$store.state.amountValue.split('.')[1].length > 1)) {
        $event.preventDefault();
      }

      if (this.$store.state.amountValue != null && this.$store.state.amountValue.toString().indexOf('.')
       > -1 && (this.$store.state.amountValue.split('.')[1].length > 1)) {
        $event.preventDefault();
      }
    },

    currencyRestrictions($event) {
      const keyCode = ($event.keyCode ? $event.keyCode : $event.which);
      if (keyCode > 47 && keyCode < 58) {
        $event.preventDefault();
      }
    },

    swapValues() {
      [this.$store.state.baseCurrency, this.$store.state.targetCurrency] = [
        this.$store.state.targetCurrency, this.$store.state.baseCurrency];
      this.$store.state.loaded = false;
    },

    setAmountValue(e) {
      this.$store.commit('setAmountValue', e.target.value);
    },

    setBaseCurrency(val) {
      this.$store.commit('setBaseCurrency', val);
      this.$store.state.loaded = false;
    },

    setTargetCurrency(val) {
      this.$store.commit('setTargetCurrency', val);
      this.$store.state.loaded = false;
    },
  },
};
</script>

<style lang="scss">
@import './SelectMenu.scss';
@import './vSelect.scss';
</style>
