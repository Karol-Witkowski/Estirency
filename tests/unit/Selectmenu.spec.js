import { expect } from 'chai';
import { shallowMount } from '@vue/test-utils';
import SelectMenu from '@/components/selectmenu/SelectMenu.vue';

describe('Home.vue test', () => {
  const wrapper = shallowMount(SelectMenu);

  it('Check the name of home view and that 3 child components are rendered', () => {
    expect(wrapper.name()).to.match(/Home/);
    expect(wrapper.findAll('.v-select').exists());
  });
});
