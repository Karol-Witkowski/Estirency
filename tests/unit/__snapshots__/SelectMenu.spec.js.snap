// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectMenu.vue test Render correctly 1`] = `
<section>
  <header>
    <h3>menu</h3>
  </header>
  <h4>Select currencies and fill input to show data</h4>
  <h4>select base currency</h4>
  <div dir="auto" class="v-select vs--single vs--searchable">
    <div id="vs1__combobox" role="combobox" aria-expanded="false" aria-owns="vs1__listbox" aria-label="Search for option" class="vs__dropdown-toggle">
      <div class="vs__selected-options"><span class="vs__selected">
            Polish zloty
           <!----></span> <input name="baseCurrency" aria-autocomplete="list" aria-labelledby="vs1__combobox" aria-controls="vs1__listbox" type="search" autocomplete="off" class="vs__search"></div>
      <div class="vs__actions"><button type="button" title="Clear Selected" aria-label="Clear Selected" class="vs__clear"><svg xmlns="http://www.w3.org/2000/svg" width="10" height="10">
            <path d="M6.895455 5l2.842897-2.842898c.348864-.348863.348864-.914488 0-1.263636L9.106534.261648c-.348864-.348864-.914489-.348864-1.263636 0L5 3.104545 2.157102.261648c-.348863-.348864-.914488-.348864-1.263636 0L.261648.893466c-.348864.348864-.348864.914489 0 1.263636L3.104545 5 .261648 7.842898c-.348864.348863-.348864.914488 0 1.263636l.631818.631818c.348864.348864.914773.348864 1.263636 0L5 6.895455l2.842898 2.842897c.348863.348864.914772.348864 1.263636 0l.631818-.631818c.348864-.348864.348864-.914489 0-1.263636L6.895455 5z"></path>
          </svg></button> <svg xmlns="http://www.w3.org/2000/svg" width="14" height="10" role="presentation" class="vs__open-indicator">
          <path d="M9.211364 7.59931l4.48338-4.867229c.407008-.441854.407008-1.158247 0-1.60046l-.73712-.80023c-.407008-.441854-1.066904-.441854-1.474243 0L7 5.198617 2.51662.33139c-.407008-.441853-1.066904-.441853-1.474243 0l-.737121.80023c-.407008.441854-.407008 1.158248 0 1.600461l4.48338 4.867228L7 10l2.211364-2.40069z"></path>
        </svg>
        <div class="vs__spinner" style="display: none;">Loading...</div>
      </div>
    </div>
    <transition-stub name="vs__fade">
      <ul id="vs1__listbox" role="listbox" style="display: none; visibility: hidden;"></ul>
    </transition-stub>
  </div>
  <figure><img alt="arrays icon" data-cy="swapIcon" src="@/assets/swap-vertical.png"></figure>
  <h4>select target currency</h4>
  <div dir="auto" class="v-select vs--single vs--searchable">
    <div id="vs2__combobox" role="combobox" aria-expanded="false" aria-owns="vs2__listbox" aria-label="Search for option" class="vs__dropdown-toggle">
      <div class="vs__selected-options"><span class="vs__selected">
            European Euro
           <!----></span> <input name="targetCurrency" aria-autocomplete="list" aria-labelledby="vs2__combobox" aria-controls="vs2__listbox" type="search" autocomplete="off" class="vs__search"></div>
      <div class="vs__actions"><button type="button" title="Clear Selected" aria-label="Clear Selected" class="vs__clear"><svg xmlns="http://www.w3.org/2000/svg" width="10" height="10">
            <path d="M6.895455 5l2.842897-2.842898c.348864-.348863.348864-.914488 0-1.263636L9.106534.261648c-.348864-.348864-.914489-.348864-1.263636 0L5 3.104545 2.157102.261648c-.348863-.348864-.914488-.348864-1.263636 0L.261648.893466c-.348864.348864-.348864.914489 0 1.263636L3.104545 5 .261648 7.842898c-.348864.348863-.348864.914488 0 1.263636l.631818.631818c.348864.348864.914773.348864 1.263636 0L5 6.895455l2.842898 2.842897c.348863.348864.914772.348864 1.263636 0l.631818-.631818c.348864-.348864.348864-.914489 0-1.263636L6.895455 5z"></path>
          </svg></button> <svg xmlns="http://www.w3.org/2000/svg" width="14" height="10" role="presentation" class="vs__open-indicator">
          <path d="M9.211364 7.59931l4.48338-4.867229c.407008-.441854.407008-1.158247 0-1.60046l-.73712-.80023c-.407008-.441854-1.066904-.441854-1.474243 0L7 5.198617 2.51662.33139c-.407008-.441853-1.066904-.441853-1.474243 0l-.737121.80023c-.407008.441854-.407008 1.158248 0 1.600461l4.48338 4.867228L7 10l2.211364-2.40069z"></path>
        </svg>
        <div class="vs__spinner" style="display: none;">Loading...</div>
      </div>
    </div>
    <transition-stub name="vs__fade">
      <ul id="vs2__listbox" role="listbox" style="display: none; visibility: hidden;"></ul>
    </transition-stub>
  </div>
  <h4>type amount</h4> <input name="amount" placeholder="amount of currency" class="amountInput">
</section>
`;
